src/shared/hooks/useNetwork.ts

```diff
@@ -0,0 +1,65 @@
+import { useMutation, useQuery } from '@tanstack/react-query';
+import axios from 'axios';
+
+import URL from 'shared/config/urls';
+
+// 파라미터 타입
+type Parameter = {
+  method: string;
+  url: string;
+  params?: object;
+};
+
+// mutation 파라미터 타입
+type MutationParameter = {
+  method: string;
+  url: string;
+  data?: object;
+};
+
+// axios 인스턴스 생성
+const axiosInstance = axios.create({
+  baseURL: URL.BASEURL,
+  timeout: 5000, // 임의로 지정
+  headers: {
+    'Content-Type': 'application/json',
+  },
+});
+
+//GET
+const fetchData = async ({ method, url, params }: Parameter) => {
+  const response = await axiosInstance({
+    method,
+    url,
+    params,
+  });
+  return response.data;
+};
+
+function useNetwork({ method, url, params }: Parameter) {
+  const queryKey = ['network', { method, url, params }];
+
+  return useQuery({
+    queryKey,
+    queryFn: () => fetchData({ method, url, params }),
+    enabled: Boolean(url),
+  });
```

Comment:
공통 config를 적용하면 조을것같아여!

```suggestion
  return useQuery({
    queryKey,
    queryFn: () => fetchData({ method, url, params }),
    retry: 3,
    // ...
    {...queryConfig} // 나머지 설정값은 props로 받으면 확장성이 더 좋아질 것 같네요~ 
  });
```